docs/book/src/migration/v1vsv2.md

```diff
@@ -0,0 +1,61 @@
+# Kubebuilder v1 vs v2
+
+There are a number of changes in
+[controller-runtime](https://github.com/kubernetes-sigs/controller-runtime),
+[controller-tools](https://github.com/kubernetes-sigs/controller-tools)
+and [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder).
+
+## Common changes
+
+- They have migrated from `dep` to `go modules` for managing dependencies.
+
+## Controller-runtime
+
+- Automatic certificate generation for webhooks has been removed, and webhooks
+will no longer self-register. Use controller-tools to generate a webhook
+configuration, and use something like cert-manager to generate your certificates
+(if you don't already have a solution). Kubebuilder supports them out-of-the-box.
+
+- If your API implements
+[`webhook.Defaulter`](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/webhook#Defaulter)
+and [`webhook.Validator`](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/webhook#Validator),
+a webhook server will be automatically wired for you.
+The tutorial can be found at [here](https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html).
+
+- Add CRD conversion webhook support. The tutorial can be found [here](https://github.com/kubernetes-sigs/kubebuilder/pull/851).
+
+- `manager` now supports adding runables that is either managed by leader election
+or always running. It depends on how runable implements
+[LeaderElectionRunable](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/manager#LeaderElectionRunnable)
+interface.
+
+- `client` package:
+  - `client.List` now uses functional options.
+  - Support delete collection.
+  - Add minimal server-side apply support. [#435](https://github.com/kubernetes-sigs/controller-runtime/pull/435)
+
+- `builder` package now has 2 sets of builders for controller and webhook respectively.
+
+- metrics are now on by default.
+
+- A number of packages under pkg/runtime have been moved, with their old locations
+deprecated.
+
+## Controller-tools
+
+The generator framework has been rewritten in v2. We tried our best to preserve the old behavior, but there some changes.
```

Comment:
Did we? That was kind-of a non-goal in many cases for this particular change, and I don't want to send the wrong message.

This is actually important, since some markers changed how they behave.